mapie/regression/ccp_regression.py

```diff
@@ -0,0 +1,806 @@
+        ] = None,
+        phi: Optional[PhiFunction] = None,
+        cv: Optional[
+            Union[str, BaseCrossValidator, BaseShuffleSplit]
+        ] = "split",
+        alpha: float = 0.1,
+        conformity_score: Optional[ConformityScore] = None,
+        random_state: Optional[int] = None,
+    ) -> None:
+
+        self.random_state = random_state
+        self.cv = cv
+        self.estimator = estimator
+        self.conformity_score_ = conformity_score
+
+        if phi is None:
+            self.phi = PhiFunction(lambda X: np.ones(len(X)))
+        else:
+            self.phi = cast(PhiFunction, phi)
+
+        self.alpha = alpha
+        self.beta_up: Optional[Tuple[NDArray, bool]] = None
+        self.beta_low: Optional[Tuple[NDArray, bool]] = None
+
+    def _check_fit_parameters(self) -> None:
+        """
+        Check and replace default ``cv`` and ``estimator`` arguments
+        """
+        self.cv = self._check_cv(
+            self.cv, random_state=self.random_state
+        )
+        self.estimator = self._check_estimator(self.estimator)
+
+    def _check_calibrate_parameters(self) -> None:
+        """
+        Check and replace default ``conformity_score`` and ``alpha`` arguments
+        """
+        self.conformity_score_ = check_conformity_score(
+            self.conformity_score_, self.default_sym_
+        )
+        self.alpha = cast(float, self._check_alpha(self.alpha))
+
+    def _check_cv(
+        self,
+        cv: Optional[Union[str, BaseCrossValidator, BaseShuffleSplit]] = None,
+        test_size: float = 0.3,
+        random_state: Optional[int] = None,
+    ) -> Union[str, BaseCrossValidator, BaseShuffleSplit]:
+        """
+        Check if ``cv`` is ``None``, ``"prefit"``, ``"split"``,
+        or ``BaseShuffleSplit``/``BaseCrossValidator`` with ``n_splits``=1.
+        Return a ``ShuffleSplit`` instance ``n_splits``=1
+        if ``None`` or ``"split"``.
+        Else raise error.
+
+        Parameters
+        ----------
+        cv: Optional[Union[str, BaseCrossValidator, BaseShuffleSplit]]
+            Cross-validator to check, by default ``None``.
+
+        test_size: float
+            If float, should be between 0.0 and 1.0 and represent the
+            proportion of the dataset to include in the test split.
+            If cv is not ``"split"``, ``test_size`` is ignored.
+
+            By default ``None``.
+
+        random_state: Optional[int]
+            Pseudo random number generator state used for random uniform
+            sampling for evaluation quantiles and prediction sets.
+            Pass an int for reproducible output across multiple function calls.
+
+            By default ```None``.
+
+        Returns
+        -------
+        Union[str, BaseCrossValidator, BaseShuffleSplit]
+            The cast `cv` parameter.
+
+        Raises
+        ------
+        ValueError
+            If the cross-validator is not valid.
+        """
+        if random_state is None:
```

Comment:
I don't use the sklearn ``check_random_state`` function, because I want to keep the ``int``, and not convert it into a ``RandomState``, for the reasons explained [here](https://github.com/scikit-learn-contrib/MAPIE/pull/455/files/7f2cf5600ae712833ca3ff3d20897b24c68c0ebd#r1626029712)